s3/pkg/datastore/yig/utils/global_id.go

```diff
@@ -0,0 +1,74 @@
+package utils
```

Comment:
can we use this instead ? BSD2 license, looks good to me
https://github.com/bwmarrin/snowflake

Reply:
thanks @rhsakarpos for your suggestion, please trust me that i have searched the similar implementation before, but the reason for why i implement it directly is that the algo is simple itself and we don't need to depends on the new 3rd lib, and also, we can customize it for that we may not need to transfer a 'machine id' when use it.